packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx

```diff
@@ -18,6 +26,29 @@ export const MockDisclosure: React.FC<{
   );
 };
 
+const useVisibleDisclosureState = (): DisclosureStateReturn => {
+  const disclosure = useDisclosureState({visible: true});
+  return {
+    ...disclosure,
+    toggle: () => {
+      disclosure.toggle();
+    },
+  };
+};
+
+const StateHookMock: React.FC = () => {
+  return (
+    <>
+      <Disclosure variant="contained" state={useVisibleDisclosureState()}>
```

Comment:
The only way React has to distinguish hooks is if a function starts with the word `use`, so let's prefer to keep these outside the `return` and placed at the top of the component closure. This makes it more clear that it isn't an ordinary function call but a React Hook behavior to future maintainers.